lib/ConstExtract/ConstExtract.cpp

```diff
@@ -128,69 +128,116 @@ static std::string extractLiteralOutput(Expr *expr) {
   return LiteralOutput;
 }
 
-static std::shared_ptr<CompileTimeValue>
-extractPropertyInitializationValue(VarDecl *propertyDecl) {
-  auto binding = propertyDecl->getParentPatternBinding();
-  if (binding) {
-    auto originalInit = binding->getOriginalInit(0);
-    if (originalInit) {
-      auto literalOutput = extractLiteralOutput(originalInit);
-      if (!literalOutput.empty()) {
-        return std::make_shared<RawLiteralValue>(literalOutput);
-      }
+static std::shared_ptr<CompileTimeValue> extractCompileTimeValue(Expr *expr) {
+  switch (expr->getKind()) {
+  case ExprKind::Array:
+  case ExprKind::Dictionary:
+  case ExprKind::BooleanLiteral:
+  case ExprKind::IntegerLiteral:
+  case ExprKind::FloatLiteral:
+  case ExprKind::NilLiteral:
+  case ExprKind::StringLiteral: {
+    std::string literalOutput;
+    llvm::raw_string_ostream OutputStream(literalOutput);
+    expr->printConstExprValue(&OutputStream, nullptr);
+
+    if (literalOutput.empty()) {
+      return std::make_shared<RuntimeValue>();
+    }
 
-      if (auto callExpr = dyn_cast<CallExpr>(originalInit)) {
-        if (callExpr->getFn()->getKind() != ExprKind::ConstructorRefCall) {
-          return std::make_shared<RuntimeValue>();
-        }
+    return std::make_shared<RawLiteralValue>(literalOutput);
+  }
 
-        std::vector<FunctionParameter> parameters;
-        const auto args = callExpr->getArgs();
-        for (auto arg : *args) {
-          auto label = arg.getLabel().str().str();
-          auto expr = arg.getExpr();
+  case ExprKind::Tuple: {
+    auto tupleExpr = cast<TupleExpr>(expr);
 
-          switch (expr->getKind()) {
-          case ExprKind::DefaultArgument: {
-            auto defaultArgument = cast<DefaultArgumentExpr>(expr);
-            auto *decl = defaultArgument->getParamDecl();
+    std::vector<TupleElement> elements;
+    if (tupleExpr->hasElementNames()) {
+      for (auto pair :
+           llvm::zip(tupleExpr->getElements(), tupleExpr->getElementNames())) {
```

Comment:
There is an assumption here that when `tupleExpr->hasElementNames()` is true, `getElements()` and `getElementNames()` have the same number of items. I can add a check, but thought it was currently unnecessary.